scripts/tests/test_check_serial_nextest_matrix.py

```
@@ -0,0 +1,333 @@
+            )
+
+    def test_reports_missing_separator(self) -> None:
+        """A runner table without a separator row should be rejected."""
+        malformed_table = "\n".join([
+            "| Runner | `#[serial]` effect | Cross-process exclusivity |",
+            (
+                "| `cargo test` | In-process mutex; required | "
+                "Not provided by `#[serial]` |"
+            ),
+            "",
+        ])
```

Ruff (0.15.20)

[warning] Consider f-string instead of string join

Replace with f-string

(FLY002)